### PR TITLE
Issue 2

### LINKED ISSUE #2: [Header] Implement and style header
- [x] Desktop version

Screenshot 2024-07-12 at 16 18 17

- [x] Mobile Version
Screenshot 2024-07-12 at 16 25 03
Screenshot 2024-07-12 at 16 25 20

Functional Requirements
- [x] Nav and Mobile Nav fragment must be fetched based on the `navpath` and `mobilepath`
- [x] Nav and Mobile Nav fragment should be fetched based on the current breakpoint, this must react to breakpoint changes (e.g. orientation changes) (use `window.matchMedia()`s `change` event)

### PR DESCRIPTION
**Pull request by Sk-Zahid**
Fix #2 

Test URLs:
- Before: https://main--piramal--aemsites.hlx.page
- After: https://issue-2--piramal--aemsites.hlx.page

### CODE REVIEW

**Comment by asthabh23** [comment]
@Sk-Zahid , is this still in progress?

few observations so far:
1. The width of the dropdown is not full screen as on original site
Screenshot 2024-08-02 at 11 21 22

2. mobile view is broken
Screenshot 2024-08-02 at 11 22 12

**Comment by Sk-Zahid** [comment]
> @Sk-Zahid , is this still in progress?
> 
> few observations so far:
> 
> 1. The width of the dropdown is not full screen as on original site
> 
> Screenshot 2024-08-02 at 11 21 22
> 2. mobile view is broken
> 
> Screenshot 2024-08-02 at 11 22 12

Please reload the page for mobile device

**Review by asthabh23 — CHANGES_REQUESTED** [review]
Below items still need a fix:

- [x] At around 1050px, the header content is getting hidden/cut from right end.

Screenshot 2024-08-05 at 12 48 56

- [x] Responsiveness needs to be handled (when we switch or reduce window width to go from desktop -> tablet -> mobile view)

Screenshot 2024-08-05 at 12 50 15

In mobile view: 
- [x] On click of hamburger icon, the entire menu should appear with an animation/transition effect from left (as on original site)
- [x] on clicking individual items in mobile menu, the subitems list should appear with a vertical animation effect. 
- [x] The selected item and its subitems also need to have a different background color (as on original site). Please see closely, the entire mobile menu isn't white (varies based on item selections)

File: styles/styles.css [hunk 1]
```diff
@@ -264,6 +267,10 @@ main pre {
   white-space: pre;
 }
 
+.dp-none {
+  display: none !important;
```
Comment (asthabh23) [inline styles/styles.css:271]: prefer not to use `!important`

File: blocks/header/header.js [hunk 2]
```diff
@@ -178,13 +194,32 @@ export default async function decorate(block) {
...
+              navSection.classList.add('active');
+              navSections.classList.add('active');
+              prevNavSection = navSection;
+            }
+          } else {
+            navSections.classList.remove('active');
+            navSection.classList.remove('active');
+            body.classList.remove('modal-open');
+            prevNavSection?.classList.remove('active');
+          }
```
Comment (asthabh23) [inline blocks/header/header.js:222]: prefer to use 'aria-expanded'/ 'aria-hidden' (to show/hide content) true/false for accessibility purpose.

File: scripts/scripts.js [hunk 3]
```diff
@@ -67,6 +67,11 @@ async function loadFonts() {
   }
 }
 
+export const targetObject = {
+  isMobile: window.matchMedia('(max-width: 767px)').matches,
+  isTab: window.matchMedia('(max-width: 1024px)').matches,
```
Comment (asthabh23) [inline scripts/scripts.js:72]: there isnt confirmation from customer so far 
we have 768px (original site) mapped to 900px for us and
1024px (original site) mapped to 1200px for us

**Review by asthabh23 — CHANGES_REQUESTED** [review]

**Review by asthabh23 — APPROVED** [review]
approving, Adobe team to complete this and incorporate the suggestions

File: styles/styles.css (hunk 1 above)
Comment (Sk-Zahid) [reply styles/styles.css]: done

File: blocks/header/header.js (hunk 2 above)
Comment (Sk-Zahid) [reply blocks/header/header.js]: done

File: scripts/scripts.js (hunk 3 above)
Comment (Sk-Zahid) [reply scripts/scripts.js]: done

**Comment by Sk-Zahid** [comment]
1. The selected item and its subitems also need to have a different background color (as on original site). Please see closely, the entire mobile menu isn't white (varies based on item selections)

This seems to resolve if other any changes please lets know

**Comment by Sk-Zahid** [comment]
Hi @asthabh23  @mind-adobe @buuhuu ,
Please review the code and update us.

**Review by asthabh23 — APPROVED** [review]

**Pull request merged**